example/http docker-compose example does not build according to instructions

From a clean checkout

```
git clone https://github.com/open-telemetry/opentelemetry-go
cd opentelemetry-go/examples/http
docker-compose up --detach http-server http-client
```

results in a failed build

```
Building http-server
Step 1/6 : FROM golang:alpine AS base
 ---> c7330979841b
Step 2/6 : COPY . /go/src/github.com/open-telemetry/opentelemetry-go/
 ---> a97d282f1af8
Step 3/6 : WORKDIR /go/src/github.com/open-telemetry/opentelemetry-go/example/http/
 ---> Running in 86f1769329c0
Removing intermediate container 86f1769329c0
 ---> 4f6a342def53

Step 4/6 : FROM base AS example-http-server
 ---> 4f6a342def53
Step 5/6 : RUN go install ./server/server.go
 ---> Running in f3344d1f2a22
server/server.go:22:2: cannot find package "go.opentelemetry.io/otel/api/correlation" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/api/correlation (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/api/correlation (from $GOPATH)
server/server.go:23:2: cannot find package "go.opentelemetry.io/otel/api/global" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/api/global (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/api/global (from $GOPATH)
server/server.go:24:2: cannot find package "go.opentelemetry.io/otel/api/trace" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/api/trace (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/api/trace (from $GOPATH)
server/server.go:25:2: cannot find package "go.opentelemetry.io/otel/exporters/trace/stdout" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/exporters/trace/stdout (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/exporters/trace/stdout (from $GOPATH)
server/server.go:26:2: cannot find package "go.opentelemetry.io/otel/plugin/httptrace" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/plugin/httptrace (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/plugin/httptrace (from $GOPATH)
server/server.go:27:2: cannot find package "go.opentelemetry.io/otel/sdk/trace" in any of:
        /usr/local/go/src/go.opentelemetry.io/otel/sdk/trace (from $GOROOT)
        /go/src/go.opentelemetry.io/otel/sdk/trace (from $GOPATH)
ERROR: Service 'http-server' failed to build: The command '/bin/sh -c go install ./server/server.go' returned a non-zero code: 1
```

Perhaps a better approach would be to use something like `GOOS=linux CGO_ENABLED=0 go build ..` to create the binaries in the development environment, and then copy them into the alpine container rather than trying to do the go build inside the alpine image (where tools such as git don't exist, and the version of go is usually an older one: e.g. 1.12.x as I write this)

--- a/example/http/Dockerfile
+++ b/example/http/Dockerfile
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM golang:alpine AS base
+FROM golang:1.14-alpine AS base
 COPY . /go/src/github.com/open-telemetry/opentelemetry-go/
 WORKDIR /go/src/github.com/open-telemetry/opentelemetry-go/example/http/
 

--- a/example/zipkin/Dockerfile
+++ b/example/zipkin/Dockerfile
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM golang:alpine
+FROM golang:1.14-alpine
 COPY . /go/src/github.com/open-telemetry/opentelemetry-go/
 WORKDIR /go/src/github.com/open-telemetry/opentelemetry-go/example/zipkin/
 RUN go install ./main.go

